pkg/local_workflows/code_workflow_test.go

```diff
@@ -0,0 +1,43 @@
+package localworkflows
+
+import (
+	"log"
+	"os"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/snyk/go-application-framework/pkg/configuration"
+	"github.com/snyk/go-application-framework/pkg/mocks"
+	"github.com/snyk/go-application-framework/pkg/workflow"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_Code_codeWorkflowEntryPoint_happyPath(t *testing.T) {
+	// set
+	logger := log.New(os.Stderr, "test", 0)
+	config := configuration.New()
+	engine := mocks.NewMockEngine(gomock.NewController(t))
```

Comment:
note: another nice way to mock is to use the normal engine, register a test workflow under the ID legacycli and in this workflow do your validations. See [here](https://github.com/snyk/go-application-framework/blob/main/pkg/workflow/engine_test.go) as an example.

Reply:
What would be the advantages of that approach?

Reply:
In general I think both are valid approaches to mock. 

The current solution needs to specify  mock behaviour for calls that are not actually of interest for the actual function under test, like the `invocationContextMock`, it exposes some implementation detail to the test, that the test doesn't need. Which can make maintainability more complex. 

Using a real engine and registering an artificial callback uses the real engine implementation and just mocks the `legacycli` implementation. Which is probably would your main interest is in this test and keeps the test more focussed.